tests/unit/testsuite/Bolt/Boltpay/Model/OrderTest.php

```diff
@@ -157,4 +157,197 @@ public function testActivateOrder()
         $this->assertEquals('TEST-BOLT-REFE-TWO2', $payment->getAdditionalInformation('bolt_reference'));
         ////////////////////////////////////////////////////////////
     }
+
+    /**
+     * @test
+     * That when the discount price is the same at Bolt and Magento, no exception is thrown
+     *
+     * @covers ::validateTotals
+     */
+    public function validateTotals_discountAccurate() {
+
+        $mocks = $this->validateTotalsSetup(25, .75, .50, 1);
+
+        $mockBoltHelper = $mocks['mockBoltHelper'];
+        $mockImmutableQuote = $mocks['mockImmutableQuote'];
+        $mockTransaction = $mocks['mockTransaction'];
+
+        $mockBoltHelper->expects($this->never())->method('logWarning');
+
+        TestHelper::callNonPublicFunction(
+            $this->testClassMock,
+            'validateTotals',
+            [$mockImmutableQuote, $mockTransaction]
+        );
+    }
+
+    /**
+     * @test
+     * That when the discount price is different at Bolt than Magento but within the price fault tolerance,
+     * no exception is thrown
+     *
+     * @covers ::validateTotals
+     */
+    public function validateTotals_discountDiffersWithinTolerance() {
+
+        $mocks = $this->validateTotalsSetup(24, .75, .50, 1);
+
+        $mockBoltHelper = $mocks['mockBoltHelper'];
+        $mockImmutableQuote = $mocks['mockImmutableQuote'];
+        $mockTransaction = $mocks['mockTransaction'];
+
+        $mockBoltHelper->expects($this->once())->method('logWarning');
+
+        TestHelper::callNonPublicFunction(
+            $this->testClassMock,
+            'validateTotals',
+            [$mockImmutableQuote, $mockTransaction]
+        );
+    }
+
+    /**
+     * @test
+     * That when the discount price is different at Bolt than Magento beyond the price fault tolerance,
+     * a general discount mismatch exception is thrown
+     *
+     * @covers ::validateTotals
+     */
+    public function validateTotals_discountDiffersBeyondTolerance() {
+
+        $mocks = $this->validateTotalsSetup(20, .75, .50, 1);
+
+        $mockBoltHelper = $mocks['mockBoltHelper'];
+        $mockImmutableQuote = $mocks['mockImmutableQuote'];
+        $mockTransaction = $mocks['mockTransaction'];
+
+        $mockBoltHelper->expects($this->never())->method('logWarning');
+
+        try {
```

Comment:
@mccottry I'd much rather be using the `expectException` workflow - can we switch over to that workflow?
https://thephp.cc/news/2016/02/questioning-phpunit-best-practices

Reply:
@matt-thomason I am definitely aware of the `expectException` method and workflow and considered that first.  Unfortunately, that method was introduced in PHPUnit 5.2 and half of our created benchmarks use PHPUnit 4.8.36, so that method is not an option.

The next "modern" approach would be to use the annotation `@expectedException` and its family.  My problem with this approach is that it does not give control over the timing in which the exception is thrown.  While this may be a minor concern in this situation, it definitely gave me pause to think.  If not using the older return and fail, this is the route that I was thinking of going.

The third consideration was the older, `setExpectedException` method and it's family.  I chose not to use that as they are deprecated in the other PHPUnit suite that we use, PHPUnit 5.7.9.  

So, no matter which way we go here, we are caught between rocks and a hard places.

1.) Uglier code - where we are now (catch, return, fail) - (pro - Exact code and handling)
2.) Code that won't run in 4.8 - `expectException` - (pro - the most elegant)
3.) Unpredictable timing on when watching exceptions - annotations - (pro-they seem cooler)
4.) Deprecated methods in 5.7.9 - `setExpectedException` - (pro - similar to expectException)

Which path do you suggest we take?